tinyxml2.cpp

```diff
@@ -75,10 +75,12 @@ distribution.
 				const int required = _vsnprintf(str, len, format, va);
 				delete[] str;
 				if ( required != -1 ) {
+					TIXMLASSERT( required >= 0 );

```

Comment:
I think that this'd make more sense if it were outside the branch? (before `if ( required != -1 )`